src/Normalizer/QueryParamsNormalizer.php

```diff
@@ -0,0 +1,64 @@
+<?php
+namespace ElevenLabs\Api\Normalizer;
+
+class QueryParamsNormalizer
+{
+    /**
+     * Normalize parameters
+     *
+     * @param array $queryParams An array of query parameters
+     * @param \stdClass $queryParamsSchema A JSON Schema of query params
+     *
+     * @return array An array of query parameters with the proper types
+     */
+    public static function normalize(array $queryParams, \stdClass $queryParamsSchema)
+    {
+        foreach ($queryParamsSchema->properties as $name => $queryParamSchema) {
+            $type = isset($queryParamSchema->type)
+                ? $queryParamSchema->type
+                : 'string';
+
+            if (array_key_exists($name, $queryParams)) {
+                switch ($type) {
+                    case 'boolean':
+                        if ($queryParams[$name] === 'false') {
+                            $queryParams[$name] = false;
+                        }
+                        if ($queryParams[$name] === 'true') {
+                            $queryParams[$name] = true;
+                        }
+                        if (in_array($queryParams[$name], ['0', '1'])) {
+                            $queryParams[$name] = (bool) $queryParams[$name];
+                        }
+                        break;
+                    case 'integer':
+                        if (is_numeric($queryParams[$name])) {
+                            $queryParams[$name] = (int) $queryParams[$name];
+                        }
+                        break;
+                    case 'number':
+                        if (is_numeric($queryParams[$name])) {
+                            $queryParams[$name] = (float) $queryParams[$name];
+                        }
+                        break;
```

Comment:
In that case, the default close is not required since other query schema types are perfectly valid. The switch case was made to apply conversion on scalar types.